### PR TITLE
Rework of LayerNorm

### PR DESCRIPTION
**Pull request by glkarpov**
Added methods to / from torch
Changed test to the up-to-date version with forward dynamic being separate test.

### CODE REVIEW

File: wrappers/python/nntile/layer/layer_norm.py [hunk 1]
```diff
@@ -339,3 +341,37 @@ def backward_async(self):
...
         # dX can offloade from GPU
         self.x.grad.wont_use()
+
+    @classmethod
+    def from_torch(cls,
+        torch_layer, x: TensorMoments,
+        axis: int, eps: float,
+        next_tag: int, redux: bool = False
+    ):
+        nntile_layer, next_tag = cls.generate_simple(x, axis, eps**2, next_tag, redux)
```
Comment (Muxas) [inline wrappers/python/nntile/layer/layer_norm.py:351]: `eps` must be read from `torch_layer`. Be careful with `eps`, as it is square rooted at line 74. `axis` shall also be deduced automatically without passing it through parameters of `from_torch` method. We assume `LayerNorm` takes as input `(n_batch, n_seq, n_emb)` tensor and, therefore, `axis` shall be `0`.

File: wrappers/python/tests/layer/test_layer_norm.py [hunk 2]
```diff
@@ -19,149 +21,194 @@
...
-    # NNTile backward of LayerNorm
-    nntile_layer.y.grad.from_array(np_B_grad)
+
+    x_shape = [params.n_size, params.m_size]
+    x_basetile = [params.n_size_tile, params.m_size_tile]
+    x_traits = TensorTraits(x_shape, x_basetile)
+    x_distr = [0] * x_traits.grid.nelems
+    x_type = dtype2nntile[dtype]
+    x_value = x_type(x_traits, x_distr, 0)
+    x_grad = x_type(x_traits, x_distr, 0)
```
Comment (Muxas) [inline wrappers/python/tests/layer/test_layer_norm.py:88]: make `x_grad=zeros_like(x_value)`. `zeros_like` is in `nntile.utils.constructors`

File: wrappers/python/tests/layer/test_layer_norm.py [hunk 3]
```diff
@@ -19,149 +21,194 @@
...
+    X = TensorMoments(x_value, x_grad, grad_required=True)
+    x_random = rng.standard_normal(x_shape)
+    x_nntile = np.array(x_random, dtype=np.float32, order="F")
+    x_value.from_array(x_nntile)
+    x_torch = torch.Tensor(x_nntile)
+    x_torch.requires_grad_()
+
+    nntile_layer, _ = nntile.layer.LayerNorm.from_torch(torch_layer, X, 1, eps, 0)
```
Comment (Muxas) [inline wrappers/python/tests/layer/test_layer_norm.py:96]: remove `axis` and `eps` from here.

File: wrappers/python/tests/layer/test_layer_norm.py [hunk 4]
```diff
@@ -19,149 +21,194 @@
...
+
+    x_shape = [params.n_size_dyn, params.m_size]
+    x_random = rng.standard_normal(x_shape)
+    x_nntile = np.array(x_random, dtype=np.float32, order="F")
+    X_other = nntile.tensor.TensorMoments(nntc.from_array(x_nntile, x_basetile_shape), None, False)
+    x_torch_other = torch.Tensor(x_nntile)
+    return torch_layer, nntile_layer, x_torch, X, x_torch_other, X_other
+
```
Comment (Muxas) [inline wrappers/python/tests/layer/test_layer_norm.py:133]: too many empty lines (2 are required)

**Review by Muxas — CHANGES_REQUESTED** [review]

**Review by Muxas — CHANGES_REQUESTED** [review]
I do not know how you managed to get it, but amount of files, changed by this PR is huge. You included many other commits, already present in main, into your PR. Make your PR show only your changes!

**Review by Muxas — COMMENTED** [review]
Add `eps=10.0` and `eps=1e-5` into test parameters objects
